packages/config/src/env/main.js

```diff
@@ -96,6 +97,23 @@ const getGeneralEnv = async function ({
   })
 }
 
+const getDeployUrls = function ({
+  siteInfo: { name = DEFAULT_SITE_NAME, ssl_url: sslUrl, build_settings: { repo_url: REPOSITORY_URL } = {} },
+  branch,
+  deployId,
+}) {
+  return {
+    URL: sslUrl,
+    REPOSITORY_URL,
+    DEPLOY_PRIME_URL: `https://${branch}--${name}${NETLIFY_DEFAULT_DOMAIN}`,
+    DEPLOY_URL: `https://${deployId}--${name}${NETLIFY_DEFAULT_DOMAIN}`,
+  }
+}
+
+const NETLIFY_DEFAULT_DOMAIN = '.netlify.app'
```

Comment:
I wonder if `NETLIFY_DEFAULT_DOMAIN` is still used when a user has a custom domain configured.
What will be the host for `DEPLOY_PRIME_URL` and `DEPLOY_URL` in that case?
If not, we might be able to take them from the `siteInfo` (if it is available)

Reply:
Good question. Yes, `.netlify.app` is still used even with a custom domain, but only for `DEPLOY_URL` and `DEPLOY_PRIME_URL`, not for `URL`. This is documented as such, but I also confirmed this with a manual test on my test site which uses a custom domain ([see build logs](https://app.netlify.com/sites/mickael-hebert/deploys/621f997199a8890008d1e9cc#L331)).